### PR TITLE
feat: find the closed color in ColorPalette

### PR DESCRIPTION
**Pull request by jczhong84**
For unknown hex tag colors, we'll try to find the closest color in the ColorPalette.

Also fixed an issue of the owner type, which needs to return a dict instead of a named tuple.

### CODE REVIEW

File: querybook/webapp/lib/utils/color.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { IColorPalette } from 'const/chartColors';
+
+const convertHexToRGB = (hexColor) => {
```
Comment (czgu) [inline querybook/webapp/lib/utils/color.ts:3]: need typing here

File: querybook/server/models/admin.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def to_dict(self):
             "id": self.id,
             "name": self.name,
             "config": loader_class.loader_config.to_dict(),
-            "owner_types": loader_class.get_table_owner_types(),
+            "owner_types": [t._asdict() for t in loader_class.get_table_owner_types()],
```
[the remark this answers is not in the document]
Comment (jczhong84) [reply querybook/server/models/admin.py:167]: this `owner_types` is nested in the metastore model's `to_dict()`, but not a direct api response. json.py will only handle the top level objects.

File: querybook/server/logic/tag.py [hunk 3]
```diff
@@ -119,7 +120,7 @@ def create_table_tags(
         meta = {
             "type": tag.type,
             "tooltip": tag.description,
-            "color": tag.color,
+            "color": closest_color(tag.color),
```
Comment (czgu) [inline querybook/server/logic/tag.py:123]: if we are assuming tag.color has format `#AABBCC` then can you add a comment in DataTag?

File: querybook/webapp/ui/Tag/Tag.tsx [hunk 4]
```diff
@@ -55,7 +55,9 @@ const StyledColorTag = styled.span.attrs<{
         let bgColor = 'var(--bg-light)';
         let textColor = 'var(--text-light)';
         if (props.color) {
-            const color = ColorPalette.find((c) => c.name === props.color);
+            const color = ColorPalette.find(
+                (c) => c.name === props.color || c.color === props.color
```
Comment (czgu) [inline querybook/webapp/ui/Tag/Tag.tsx:59]: `c.color === props.color` is not needed

**Review by czgu — APPROVED** [review]

**Pull request merged**